src/components/button/Button.tsx

```diff
@@ -53,22 +57,51 @@ class Button extends React.Component<ButtonProps> {
     };
 
     render() {
-        const { children, className, isDisabled, isLoading, isSelected, setRef, type, showRadar, ...rest } = this.props;
+        const {
+            children,
+            className,
+            icon,
+            isDisabled,
+            isLoading,
+            isSelected,
+            setRef,
+            size,
+            type,
+            showRadar,
+            ...rest
+        } = this.props;
         const buttonProps: Record<string, boolean> = omit(rest, ['onClick']);
         if (isDisabled) {
             buttonProps['aria-disabled'] = true;
         }
 
+        const hasIcon = !!icon;
+        const hasText = !!children;
+
         const styleClassName = classNames(
             'btn',
             {
                 'is-disabled': isDisabled,
                 'is-loading': isLoading,
                 'is-selected': isSelected,
+                'bdl-is-large': size === 'large',
+                'bdl-has-icon': hasIcon,
             },
             className,
         );
 
+        const textContent = children ? <span className="btn-content">{children}</span> : null;
+        let iconContent = null;
+        if (icon) {
+            // Size of text+icon is 16px, just icon is 20px
+            const iconSize = hasIcon && hasText ? 16 : 20;
```

Comment:
Does the icon need to grow for the `large` variant?

Reply:
The main use case for icon buttons is the large size. Regular size is supported with the same size per design, but not commonly used.